controller.py

```diff
@@ -310,6 +312,63 @@ def controller(
     if istart != -1:
         faultlist = faultlist[istart:]
 
+    # Remove already simulated faults
+    if missing_only:
+        clogger.info("Identifying already simulated faults")
+
+        hdf5_file = Path(hdf5path)
+        if hdf5_file.is_file():
+            # Collect list of simulated faults
+            existing_faults = []
+
+            with tables.open_file(hdf5path, "r") as f_in:
+                for group in f_in.walk_groups("/fault"):
+                    if group._v_depth != 2:
+                        continue
+
+                    fault_entry = group.faults
+
+                    assert len(fault_entry) == 1
+
+                    fa = fault_entry[0]["fault_address"]
+                    ta = fault_entry[0]["trigger_address"]
+                    th = fault_entry[0]["trigger_hitcounter"]
+                    nb = fault_entry[0]["fault_num_bytes"]
+
+                    fault_str = f"{fa:x}_{ta:x}_{th}_{nb}"
```

Comment:
This implements only a partial match of the fault properties. Please expand this to cover all properties.